Realm/RLMRealm.mm

```diff
@@ -294,6 +294,17 @@ + (NSString *)writeablePathForFile:(NSString*)fileName
     return [path stringByAppendingPathComponent:fileName];
 }
 
++ (NSString *)writeableTemporaryPathForFile:(NSString*)fileName
+{
+    NSString *path = NSTemporaryDirectory();
+    [[NSFileManager defaultManager] createDirectoryAtPath:path
+                              withIntermediateDirectories:YES
+                                               attributes:nil
+                                                    error:nil];
+    fileName = [NSString stringWithFormat:@"%d-%@", NSProcessInfo.processInfo.processIdentifier, fileName];
```

Comment:
Needs to give a consistent filename for the identifier so that an in-memory realm can be opened in multiple threads or processes (which this change currently breaks).